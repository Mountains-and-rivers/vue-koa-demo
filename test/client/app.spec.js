import { shallowMount, mount } from '@vue/test-utils'
import App from '@/App'
import Todoheader from '@/components/Todoheader'

describe('App.vue', () => {
  beforeEach(() => {
    sessionStorage.clear()
  })

  /**
   *  DATA TEST
   */
  test('data.username should be sessionStorage.username: breezymelon', () => {
    const $router = { replace: jest.fn() }
    sessionStorage.__STORE__.username = 'breezymelon'
    const wrapper = shallowMount(App, {
      mocks: {
        $router
      }
    })
    expect(wrapper.vm.username).toBe('breezymelon')
  })

  test('data.username should be empty string', () => {
    const wrapper = shallowMount(App)

    expect(wrapper.vm.username).toBe('')
  })

  test('setUserName should be called when emit setUserName event', () => {
    const wrapper = mount(App, {
      stubs: ['router-link', 'router-view']
    })
    wrapper.find(Todoheader).vm.$emit('setUserName', { username: 'breezymelon' })

    expect(wrapper.vm.username).toBe('breezymelon')
  })
})
