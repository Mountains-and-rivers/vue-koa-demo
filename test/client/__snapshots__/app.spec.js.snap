// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue app.vue snapshot test 1`] = `
<div
  id="app"
>
  <todo-header-stub
    username=""
  />
   
  <router-view-stub />
   
  <todo-footer-stub />
</div>
`;
